File: PCMania/src/Computador/HardwareBasico.java
package Computador;

public class HardwareBasico {
    private String nome;
    private float capacidade;

    // Construtor para inicializar os atributos
    public HardwareBasico(String nome, float capacidade) {
        this.nome = nome;
        this.capacidade = capacidade;
    }

    //Getters
    public String getNome() {
        return nome;
    }

    public float getCapacidade() {
        return capacidade;
    }

}

File: POO_CPP/PCManiaCPP/src/Computador/HardwareBasico.cpp
#include "HardwareBasico.h"

using namespace std;

// Implementação do Construtor
HardwareBasico::HardwareBasico(string nome, float capacidade) {
    this->nome = nome;
    this->capacidade = capacidade;
}

// Implementação do Getter para o nome
string HardwareBasico::getNome() {
    return this->nome;
}

// Implementação do Getter para a capacidade
float HardwareBasico::getCapacidade() {
    return this->capacidade;
}